config/core/deployments/autoscaler.yaml

```diff
@@ -20,7 +21,7 @@ metadata:
   labels:
     serving.knative.dev/release: devel
 spec:
-  replicas: 1
+  replicas: 0 # 0 pod as this deployment is deprecated.
```

Comment:
Should we leave this at `1` for a release to ensure we're not running into an outage where you can't scale from 0 in the upgrade?

Reply:
How can we make this to zero after the upgrade? If there are both a pod of Deployment and a pod of StatefulSet, they will compete to make scaling decisions.

Reply:
Also how to redirect activator? Use the same k8s service for a transition period?
So the options are:
- downtime while the new release is released 
- double reconciles (but this solves scale from 0 problem since any AS can act on poke), probably if the time window is short not the very worst scenario.
- and hack/shim that in SS mode watches the pod of the old deployment and while it's ready ignores reconcilications and only does things when it's the only autoscaler in the system (still need to act on pokes, I think)
 
cc @mattmoor  for more ideas/

Reply:
@markusthoemmes we have a scale from 0 outage during every upgrade today, right?

Reply:
> * double reconciles (but this solves scale from 0 problem since any AS can act on poke), probably if the time window is short not the very worst scenario.

How do we set Autoscaler Deployment replica to 0 after Autoscaler StatefulSet pod is up. Does kubectl guarantee the order?

> * and hack/shim that in SS mode watches the pod of the old deployment and while it's ready ignores reconcilications and only does things when it's the only autoscaler in the system (still need to act on pokes, I think)
Looks like you're talking about keep both for next release and set deployment to 0 for the one after next release.

Reply:
> @markusthoemmes we have a scale from 0 outage during every upgrade today, right?

Doesn't deployment kill the old one only after new one is ready?

Reply:
Yeah the old one would be killed after the new one is ready today